tap_braintree/__init__.py

```diff
@@ -196,6 +199,22 @@ def sync_transactions():
     singer.write_state(STATE)
 
 
+def do_discover():
+    # Generate a client token to verify credentials
+    try:
+        braintree.ClientToken.generate()
+        logger.info("Braintree configuration is valid. Token generated.")
+    except braintree.exceptions.authentication_error.AuthenticationError:
+        logger.info("Authentication error: Check your credentials.")
+    except Exception as e:
+        logger.info(f"Configuration error: {e}")
```

Comment:
Existing code would not exit gracefully if credential validation fails. It will just log the error message as info and successfully run the discovery. Rather, it will fail during the extraction.

```suggestion
try:
    braintree.ClientToken.generate()
    logger.info("Braintree configuration is valid.")
except braintree.exceptions.authentication_error.AuthenticationError as ex:
    raise Exception("Authentication error: Check your credentials.") from ex
except Exception:
    raise Exception("Unexpected error during Braintree configuration validation.")
```

Reply:
Please review - the changes were committed